Sources/NIOEmbedded/Embedded.swift

```diff
@@ -92,16 +97,17 @@ public final class EmbeddedEventLoop: EventLoop {
     @discardableResult
     public func scheduleTask<T>(deadline: NIODeadline, _ task: @escaping () throws -> T) -> Scheduled<T> {
         let promise: EventLoopPromise<T> = makePromise()
-        let task = EmbeddedScheduledTask(readyTime: deadline, insertOrder: self.nextTaskNumber()) {
+        let task = EmbeddedScheduledTask(readyTime: deadline, insertOrder: self.nextTaskNumber(), task: {
             do {
                 promise.succeed(try task())
             } catch let err {
                 promise.fail(err)
             }
-        }
+        })
 
+        let taskId = task.id
         let scheduled = Scheduled(promise: promise, cancellationTask: {
-            self.scheduledTasks.remove(task)
+            self.scheduledTasks.removeAll { $0.id == taskId }
```

Comment:
We don't need `removeAll`, we only need a `removeFirst` function: scanning the entire array once we've found the one we want to remove is unnecessary.